instrumentation/src/main/java/io/opentelemetry/android/internal/features/persistence/DiskManager.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.features.persistence;
+
+import io.opentelemetry.android.config.OtelRumConfig;
+import io.opentelemetry.android.config.PersistenceConfiguration;
+import io.opentelemetry.android.internal.services.AppInfoService;
+import io.opentelemetry.android.internal.services.PreferencesService;
+import io.opentelemetry.android.internal.services.Service;
+import io.opentelemetry.android.internal.services.ServiceManager;
+import java.io.File;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class is internal and not for public use. Its APIs are unstable and can change at any time.
+ */
+public final class DiskManager {
+    private final AppInfoService appInfoService;
+    private final PreferencesService preferencesService;
+    private final PersistenceConfiguration persistenceConfiguration;
+    private static final String MAX_FOLDER_SIZE_KEY = "max_signal_folder_size";
+    private static final int MAX_FILE_SIZE = 1024 * 1024;
+    private final static Logger logger = Logger.getLogger("DiskManager");
+
+    public static DiskManager create(OtelRumConfig config) {
+        ServiceManager serviceManager = ServiceManager.get();
+        return new DiskManager(
+                serviceManager.getService(Service.Type.APPLICATION_INFO),
+                serviceManager.getService(Service.Type.PREFERENCES),
+                config.getPersistenceConfiguration());
+    }
+
+    DiskManager(
+            AppInfoService appInfoService,
+            PreferencesService preferencesService,
+            PersistenceConfiguration persistenceConfiguration) {
+        this.appInfoService = appInfoService;
+        this.preferencesService = preferencesService;
+        this.persistenceConfiguration = persistenceConfiguration;
+    }
+
+    public File getSignalsCacheDir() throws IOException {
+        File dir = new File(appInfoService.getCacheDir(), "opentelemetry/signals");
+        if (!dir.exists()) {
```

Comment:
`mkdirs` already calls `exists` internally so you are doing IO twice.
To consider is that `mkdirs` returns `false` if `exists` returns true, so you cannot `throw IOException` in this case which is also suboptinal.

Reply:
`Files.deleteIfExists` does that better but its probably not available under older Android versions.

Reply:
> `Files.deleteIfExists` does that better but its probably not available under older Android versions.

Yeah, it seems like it's available from API 26 😅